src/core/common/string.cpp

```diff
@@ -72,30 +73,56 @@ bool StringEndsWith(const char *aString, char aChar)
     return len > 0 && aString[len - 1] == aChar;
 }
 
-Error StringBase::Write(char *aBuffer, uint16_t aSize, uint16_t &aLength, const char *aFormat, va_list aArgs)
+StringWriter::StringWriter(char *aBuffer, uint16_t aSize)
+    : mBuffer(aBuffer)
+    , mLength(0)
+    , mSize(aSize)
 {
-    Error error = kErrorNone;
-    int   len;
+    mBuffer[0] = 0;
+}
 
-    len = vsnprintf(aBuffer + aLength, aSize - aLength, aFormat, aArgs);
+StringWriter &StringWriter::Clear(void)
+{
+    mBuffer[0] = 0;
```

Comment:
nit: to be more explicit
```suggestion
    mBuffer[0] = '\0';
```